src/target/lpc55xx.c

```diff
@@ -37,12 +37,478 @@
+// #define LPC55S16_CHIPID           // (unknown)
+#define LPC55S26_CHIPID   0xA010229A // NXP forum
+#define LPC55S28_CHIPID   0xA010119C // NXP forum
+// #define LPC55S66_CHIPID           // (unknown)
+// #define LPC55S69_CHIPID           // (unknown)
+
+ /* The available IAP commands that we support, mostly flash access */
+
+enum iap_cmd {
+	IAP_CMD_FLASH_INIT,
+	IAP_CMD_FLASH_ERASE,
+	IAP_CMD_FLASH_PROGRAM,
+	IAP_CMD_FFR_INIT,
+	IAP_CMD_FFR_GET_UUID,
+};
+
+/* The possible IAP errors are documented here for easy reference */
+
+enum iap_status {
+	IAP_STATUS_FLASH_SUCCESS = 0,
+	IAP_STATUS_FLASH_INVALID_ARGUMENT = 4,
+	IAP_STATUS_FLASH_SIZE_ERROR = 100,
+	IAP_STATUS_FLASH_ALIGNMENT_ERROR = 101,
+	IAP_STATUS_FLASH_ADDRESS_ERROR = 102,
+	IAP_STATUS_FLASH_ACCESS_ERROR = 103,
+	IAP_STATUS_FLASH_COMMAND_FAILURE = 105,
+	IAP_STATUS_FLASH_UNKNOWN_PROPERTY = 106,
+	IAP_STATUS_FLASH_ERASE_KEY_ERROR = 107,
+	IAP_STATUS_FLASH_COMMAND_NOT_SUPPORTED = 111,
+	IAP_STATUS_FLASH_ECC_ERROR = 116,
+	IAP_STATUS_FLASH_COMPARE_ERROR = 117,
+	IAP_STATUS_FLASH_REGULATION_LOSS = 118,
+	IAP_STATUS_FLASH_INVALID_WAIT_STATE_CYCLES = 119,
+	IAP_STATUS_FLASH_OUT_OF_DATE_CFPA_PAGE = 132,
+	IAP_STATUS_FLASH_BLANK_IFR_PAGE_DATA = 133,
+	IAP_STATUS_FLASH_ENCRYPTED_REGIONS_ERASE_NOT_DONE_AT_ONCE = 134,
+	IAP_STATUS_FLASH_PROGRAM_VERIFICATION_NOT_ALLOWED = 135,
+	IAP_STATUS_FLASH_HASH_CHECK_ERROR = 136,
+	IAP_STATUS_FLASH_SEALED_FFR_REGION = 137,
+	IAP_STATUS_FLASH_FFR_REGION_WRITE_BROKEN = 138,
+	IAP_STATUS_FLASH_NMPA_ACCESS_NOT_ALLOWED = 139,
+	IAP_STATUS_FLASH_CMPA_CFG_DIRECT_ERASE_NOT_ALLOWED = 140,
+	IAP_STATUS_FLASH_FFR_BANK_IS_LOCKED = 141,
+};
+
+static target_addr_t lpc55xx_get_bootloader_tree_address(target* t)
+{
+	switch (target_mem_read32(t, LPC55XX_CHIPID_ADDRESS)) {
+	//case LPC5512_CHIPID:
+	//case LPC5514_CHIPID:
+	//case LPC55S14_CHIPID:
+	//case LPC5516_CHIPID:
+	//case LPC55S16_CHIPID:
+	//case LPC5524_CHIPID:
+	case LPC5502_CHIPID:
+	case LPC5504_CHIPID:
+	case LPC5506_CHIPID:
+	case LPC55S04_CHIPID:
+	case LPC55S06_CHIPID:
+		return 0x1301fe00;
+
+	//case LPC55S69_CHIPID:
+	case LPC5526_CHIPID:
+	case LPC55S26_CHIPID:
+	case LPC5528_CHIPID:
+	case LPC55S28_CHIPID:
+	//case LPC55S66_CHIPID:
+		return 0x130010f0;
+
+	default:
+		return 0;
+	}
+}
+
+static const char* lpc55xx_get_device_name(uint32_t chipid)
+{
+	switch (chipid) {
+	case LPC5502_CHIPID:
+		return "LPC5502";
+	case LPC5504_CHIPID:
+		return "LPC5504";
+	case LPC5506_CHIPID:
+		return "LPC5506";
+	case LPC55S04_CHIPID:
+		return "LPC55S04";
+	case LPC55S06_CHIPID:
+		return "LPC55S06";
+	case LPC5526_CHIPID:
+		return "LPC5526";
+	case LPC55S26_CHIPID:
+		return "LPC55S26";
+	case LPC5528_CHIPID:
+		return "LPC5528";
+	case LPC55S28_CHIPID:
+		return "LPC55S28";
+
+	default:
+		return "unknown";
+	}
+}
+
+static int lpc55xx_get_rom_api_version(target* t, target_addr_t bootloader_tree_address)
+{
+	return (((target_mem_read32(t, bootloader_tree_address + 0x4) >> 16) & 0xff) == 3) ? 1 : 0;
```

Comment:
The outer parens are not required so please remove them. In a similar vein, please suffix the bit manipulation constants with U